Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionFactoryTest.cs

```diff
@@ -134,5 +160,96 @@ public void CreateTypeDefinitions_WithNonClassOrInterface_Throws ()
           Throws.InvalidOperationException
               .With.Message.EqualTo("Cannot create a builder node for type 'System.TypeCode' because it is not a class-type."));
     }
+
```

Comment:
The rest of the functionality is covered by the `ClassDefinitionBuilderNode` + `InterfaceDefinitionBuilderNode`. The `TypeDefinitionFactory` is just using those correctly and as such I built the test to cover more specific use cases.

Reply:
This is such a big ball of yarn.  Even though it might be redundant, having a straight forward "Create Interface" and "Create Class + Interface" test does seem like the "correct" choice for the API. That way, typedeffactory can start its tests by doing basic stuff. It's a bit of an expectation problem: you expect certain tests for a SUT, if they're missing, it feels like problem.